public/js/saveApplicant.js

```diff
@@ -32,34 +73,50 @@ let makeApiRequest = async (data) => {
         },
         method: 'post',
         body: JSON.stringify(data)
-    })
-        .then(response => response.json())
-        .then((data) => {
-            if (data.success) {
-                window.location.href = './admin';
-            } else {
-                document.getElementById('errorMsg').innerHTML = "Please contact administrator. Or <a href='./admin'>click here</a> to return to admin page"
-            }
-        })
-}
-
-let validateFormInputs = (data) => {
-    let validate = []
+    }).then(response => {
+        const generalErrorMessage = document.querySelector('#generalError');
 
-    validate.push(evalLength(data.name))
-    validate.push(evalLength(data.email))
-    validate.push(evalLength(data.phoneNumber))
-    validate.push(evalLength(data.whyDev))
-    validate.push(evalLength(data.codeExperience))
-    validate.push(evalLength(data.hearAboutId))
+        if (response.status === 200) {
+            window.location.href = './admin';
```

Comment:
why?! this is so annoying! you submit the form and with no idea if it worked or not your suddenly on the admin panel?!